crates/rover-client/src/shared/check_response.rs

```diff
@@ -259,6 +273,77 @@ impl LintCheckResponse {
     }
 }
 
+#[derive(Debug, Serialize, Clone, Eq, PartialEq)]
+
+pub enum ProposalsCheckSeverityLevel {
+    ERROR,
+    OFF,
+    WARN,
+}
+
+#[derive(Debug, Serialize, Clone, Eq, PartialEq)]
+pub struct RelatedProposal {
+    pub status: String,
+    pub display_name: String,
+}
+
+#[derive(Debug, Serialize, Clone, Eq, PartialEq)]
+pub struct ProposalsCheckResponse {
+    pub task_status: CheckTaskStatus,
+    pub severity_level: ProposalsCheckSeverityLevel,
+    pub target_url: Option<String>,
+    pub related_proposals: Vec<RelatedProposal>,
+}
+
+impl ProposalsCheckResponse {
+    pub fn get_table(&self) -> String {
+        let mut table = Table::new();
+
+        table.set_format(*FORMAT_BOX_CHARS);
+
+        // bc => sets top row to be bold and center
+        table.add_row(row![bc =>  "Status", "Proposal Name"]);
+
+        for proposal in &self.related_proposals {
+            table.add_row(row![proposal.status, proposal.display_name,]);
+        }
+
+        table.to_string()
+    }
+
+    pub fn get_msg(&self) -> String {
+        match self.severity_level {
+            ProposalsCheckSeverityLevel::ERROR => "Your check failed because some or all of the proposals matching this change have not been approved.".to_string(),
+            ProposalsCheckSeverityLevel::WARN => "Your check passed with warnings because some or all of the proposals matching this change have not been approved.".to_string(),
```

Comment:
```suggestion
            ProposalsCheckSeverityLevel::WARN => "Your check passed with warnings because some or all of the diffs in this change are not in an approved Proposal.".to_string(),
```